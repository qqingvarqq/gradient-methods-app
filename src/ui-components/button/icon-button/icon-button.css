.icon-button {
  user-select: none;
  font-weight: 400;
}
