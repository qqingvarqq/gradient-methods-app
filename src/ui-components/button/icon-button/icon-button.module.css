.root {
  user-select: none;
  font-weight: 400;
}
