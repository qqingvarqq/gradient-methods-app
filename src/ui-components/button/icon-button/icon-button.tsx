import * as React from 'react';
import styles from './icon-button.module.css';
import {BaseButton, BaseButtonProps} from '../base-button/base-button';

export enum IconButtonTypes {
  ARROW_LEFT = 'ARROW_LEFT',
  ARROW_RIGHT = 'ARROW_RIGHT',
}

function getIcon(iconType?: IconButtonTypes): string {
  switch (iconType) {
    case IconButtonTypes.ARROW_LEFT:
      return '←';
    case IconButtonTypes.ARROW_RIGHT:
      return '→';
    default:
      return '';
  }
}

function getInlineStyles(size: string) {
  return {height: size, width: size, fontSize: size};
}

interface IconButtonProps extends BaseButtonProps {
  size?: string;
  iconType: IconButtonTypes;
}

export const IconButton: React.FC<IconButtonProps> = (
  props: IconButtonProps
) => {
  const {iconType, size = '48px', className = '', ...rest} = props;
  const icon = getIcon(iconType);
  const cssStyles = getInlineStyles(size);
  return (
    <BaseButton
      style={cssStyles}
      className={`${styles.root} ${className}`}
      {...rest}>
      {icon}
    </BaseButton>
  );
};
