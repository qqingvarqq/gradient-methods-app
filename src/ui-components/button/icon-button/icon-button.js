import React from 'react';
import './icon-button.css';
import {BaseButton} from '../base-button';

function getIcon(type) {
  switch (type) {
    case 'arrowLeft':
      return '‚Üê';

    default:
      return '';
  }
}
function getInlineStyles(size) {
  return {height: size, width: size, fontSize: size};
}

export default function IconButton(props) {
  const {type, size = '48px', className, ...rest} = props;
  const classNames = 'icon-button ' + (className || '');
  const icon = getIcon(type);
  const cssStyles = getInlineStyles(size);
  return (
    <BaseButton style={cssStyles} className={classNames} {...rest}>
      {icon}
    </BaseButton>
  );
}
