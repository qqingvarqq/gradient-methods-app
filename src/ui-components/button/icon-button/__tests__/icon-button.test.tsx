import * as React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {IconButton, IconButtonTypes} from '../icon-button';

describe('IconButton', () => {
  it('should render without crashing', () => {
    const icons = Object.values(IconButtonTypes);
    for (let icon of icons) {
      //@ts-ignore
      const {asFragment} = render(<IconButton iconType={icon} />);
      expect(asFragment()).toMatchSnapshot();
    }
  });
  it('should be clickable', () => {
    const onClick = jest.fn();
    const {container} = render(
      <IconButton iconType={IconButtonTypes.ARROW_LEFT} onClick={onClick} />
    );
    // @ts-ignore
    fireEvent.click(container.firstChild);
    expect(onClick).toBeCalled();
  });
});
