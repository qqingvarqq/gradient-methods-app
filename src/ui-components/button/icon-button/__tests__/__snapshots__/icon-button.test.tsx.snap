// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should render without crashing 1`] = `
<DocumentFragment>
  <button
    class="root root "
    style="height: 48px; width: 48px; font-size: 48px;"
  >
    ←
  </button>
</DocumentFragment>
`;

exports[`IconButton should render without crashing 2`] = `
<DocumentFragment>
  <button
    class="root root "
    style="height: 48px; width: 48px; font-size: 48px;"
  >
    →
  </button>
</DocumentFragment>
`;
